--- Lib/distutils/util.py	2011-06-14 15:27:46.627638000 +1000
+++ /tmp/tmp.HdBVp23095/src/Python-3.2/Lib/distutils/util.py	2011-06-14 15:16:53.000000000 +1000
@@ -221,7 +221,7 @@
         if not os.path.isabs(pathname):
             return os.path.join(new_root, pathname)
         else:
-            return os.path.join(new_root, pathname[1:])
+            return os.path.join(new_root, pathname.lstrip('/'))
 
     elif os.name == 'nt':
         (drive, path) = os.path.splitdrive(pathname)
