--- Python-3.5.2.orig/Lib/ctypes/util.py
+++ Python-3.5.2/Lib/ctypes/util.py
@@ -132,22 +132,7 @@
             return res.group(1)
     else:
         def _get_soname(f):
-            # assuming GNU binutils / ELF
-            if not f:
-                return None
-            cmd = 'if ! type objdump >/dev/null 2>&1; then exit 10; fi;' \
-                  "objdump -p -j .dynamic 2>/dev/null " + f
-            f = os.popen(cmd)
-            try:
-                dump = f.read()
-            finally:
-                rv = f.close()
-            if rv == 10:
-                raise OSError('objdump command not found')
-            res = re.search(r'\sSONAME\s+([^\s]+)', dump)
-            if not res:
-                return None
-            return res.group(1)
+            return os.path.basename(os.path.realpath(f))
 
     if sys.platform.startswith(("freebsd", "openbsd", "dragonfly")):
 
