--- Python-3.7.1.orig/Lib/ctypes/util.py
+++ Python-3.7.1/Lib/ctypes/util.py
@@ -157,25 +157,7 @@
     else:
         def _get_soname(f):
             # assuming GNU binutils / ELF
-            if not f:
-                return None
-            objdump = shutil.which('objdump')
-            if not objdump:
-                # objdump is not available, give up
-                return None
-
-            try:
-                proc = subprocess.Popen((objdump, '-p', '-j', '.dynamic', f),
-                                        stdout=subprocess.PIPE,
-                                        stderr=subprocess.DEVNULL)
-            except OSError:  # E.g. bad executable
-                return None
-            with proc:
-                dump = proc.stdout.read()
-            res = re.search(br'\sSONAME\s+([^\s]+)', dump)
-            if not res:
-                return None
-            return os.fsdecode(res.group(1))
+            return os.path.basename(os.path.realpath(f))
 
     if sys.platform.startswith(("freebsd", "openbsd", "dragonfly")):
 
