--- busybox-1.32.1.orig/init/halt.c
+++ busybox-1.32.1/init/halt.c
@@ -230,9 +230,22 @@
 				/* runlevels:
 				 * 0 == shutdown
 				 * 6 == reboot */
+				char *signal;
+				switch(which)
+				{
+					case 2:
+						signal = "6";
+						break;
+					case 1:
+						signal = "7";
+						break;
+					case 0:
+						signal = "0";
+						break;
+				}
 				execlp(CONFIG_TELINIT_PATH,
 						CONFIG_TELINIT_PATH,
-						which == 2 ? "6" : "0",
+						signal,
 						(char *)NULL
 				);
 				bb_perror_msg_and_die("can't execute '%s'",
