diff -Naur kmod-5-orig/libkmod/libkmod-array.c kmod-5/libkmod/libkmod-array.c
--- kmod-5-orig/libkmod/libkmod-array.c	2012-01-11 17:35:18.202581879 -0500
+++ kmod-5/libkmod/libkmod-array.c	2012-02-08 20:31:07.417593419 -0500
@@ -18,13 +18,12 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#include "libkmod.h"
-#include "libkmod-array.h"
-
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include "libkmod.h"
+#include "libkmod-array.h"
 
 /* basic pointer array growing in steps */
 
diff -Naur kmod-5-orig/libkmod/libkmod-module.c kmod-5/libkmod/libkmod-module.c
--- kmod-5-orig/libkmod/libkmod-module.c	2012-02-06 09:52:13.444034579 -0500
+++ kmod-5/libkmod/libkmod-module.c	2012-02-08 20:31:07.417593558 -0500
@@ -40,6 +40,10 @@
 #include "libkmod.h"
 #include "libkmod-private.h"
 
+# include <sys/syscall.h>
+# define init_module(mod, len, opts) syscall(__NR_init_module, mod, len, opts)
+# define delete_module(mod, flags) syscall(__NR_delete_module, mod, flags)
+
 /**
  * SECTION:libkmod-module
  * @short_description: operate on kernel modules
@@ -749,7 +753,7 @@
 }
 
 
-extern long delete_module(const char *name, unsigned int flags);
+/* extern long delete_module(const char *name, unsigned int flags); */
 
 /**
  * kmod_module_remove_module:
@@ -780,7 +784,7 @@
 	return err;
 }
 
-extern long init_module(const void *mem, unsigned long len, const char *args);
+/* extern long init_module(const void *mem, unsigned long len, const char *args); */
 
 /**
  * kmod_module_insert_module:
diff -Naur kmod-5-orig/libkmod/libkmod-util.c kmod-5/libkmod/libkmod-util.c
--- kmod-5-orig/libkmod/libkmod-util.c	2012-01-31 11:18:44.806005190 -0500
+++ kmod-5/libkmod/libkmod-util.c	2012-02-10 20:08:07.837032128 -0500
@@ -31,6 +31,19 @@
 #include "libkmod.h"
 #include "libkmod-private.h"
 
+#ifdef NO_GET_CURRENT_DIR_NAME
+char *get_current_dir_name()
+{
+	struct stat a, b;
+	char buf[PATH_MAX];
+	char *res = getenv("PWD");
+	if (res && *res && (stat(res, &a) == 0) && (stat(".", &b) == 0) &&
+	(a.st_dev == b.st_dev) && (a.st_ino == b.st_ino)) return strdup(res);
+	if (!getcwd(buf, sizeof buf)) return NULL;
+	return strdup(buf);
+}
+#endif
+
 /*
  * Read one logical line from a configuration file.
  *
diff -Naur kmod-5-orig/libkmod/libkmod-util.h kmod-5/libkmod/libkmod-util.h
--- kmod-5-orig/libkmod/libkmod-util.h	2012-01-20 13:34:00.840398110 -0500
+++ kmod-5/libkmod/libkmod-util.h	2012-02-09 23:44:32.724567249 -0500
@@ -6,6 +6,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 
 
 char *getline_wrapped(FILE *fp, unsigned int *linenum) __attribute__((nonnull(1)));
diff -Naur kmod-5-orig/tools/kmod-depmod.c kmod-5/tools/kmod-depmod.c
--- kmod-5-orig/tools/kmod-depmod.c	2012-02-06 15:00:36.502714514 -0500
+++ kmod-5/tools/kmod-depmod.c	2012-02-08 20:31:07.417593558 -0500
@@ -16,11 +16,6 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#include "libkmod.h"
-#include "libkmod-array.h"
-#include "libkmod-hash.h"
-#include "libkmod-util.h"
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <getopt.h>
@@ -36,6 +31,11 @@
 #include <unistd.h>
 #include <ctype.h>
 
+#include "libkmod.h"
+#include "libkmod-array.h"
+#include "libkmod-hash.h"
+#include "libkmod-util.h"
+
 #define DEFAULT_VERBOSE LOG_WARNING
 static int verbose = DEFAULT_VERBOSE;
 
