--- php-5.6.29.orig/acinclude.m4
+++ php-5.6.29/acinclude.m4
@@ -2679,20 +2679,20 @@
   done
 
   echo "'[$]0' \\" >> $1
-  if test `expr -- [$]0 : "'.*"` = 0; then
+  if test `expr " [$]0" : " '.*"` = 0; then
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND '[$]0'"
   else 
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND [$]0"
   fi
   for arg in $ac_configure_args; do
-    if test `expr -- $arg : "'.*"` = 0; then
-      if test `expr -- $arg : "-.*"` = 0 && test `expr -- $arg : ".*=.*"` = 0; then
+    if test `expr " $arg" : " '.*"` = 0; then
+      if test `expr " $arg" : " -.*"` = 0 && test `expr " $arg" : " .*=.*"` = 0; then
         continue;
       fi
       echo "'[$]arg' \\" >> $1
       CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS '[$]arg'"
     else
-      if test `expr -- $arg : "'-.*"` = 0 && test `expr -- $arg : "'.*=.*"` = 0; then
+      if test `expr " $arg" : " '-.*"` = 0 && test `expr " $arg" : " '.*=.*"` = 0; then
         continue;
       fi
       echo "[$]arg \\" >> $1
--- php-5.6.29.orig/aclocal.m4
+++ php-5.6.29/aclocal.m4
@@ -2679,20 +2679,20 @@
   done
 
   echo "'[$]0' \\" >> $1
-  if test `expr -- [$]0 : "'.*"` = 0; then
+  if test `expr " [$]0" : " '.*"` = 0; then
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND '[$]0'"
   else 
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND [$]0"
   fi
   for arg in $ac_configure_args; do
-    if test `expr -- $arg : "'.*"` = 0; then
-      if test `expr -- $arg : "-.*"` = 0 && test `expr -- $arg : ".*=.*"` = 0; then
+    if test `expr " $arg" : " '.*"` = 0; then
+      if test `expr " $arg" : " -.*"` = 0 && test `expr " $arg" : " .*=.*"` = 0; then
         continue;
       fi
       echo "'[$]arg' \\" >> $1
       CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS '[$]arg'"
     else
-      if test `expr -- $arg : "'-.*"` = 0 && test `expr -- $arg : "'.*=.*"` = 0; then
+      if test `expr " $arg" : " '-.*"` = 0 && test `expr " $arg" : " '.*=.*"` = 0; then
         continue;
       fi
       echo "[$]arg \\" >> $1
