From 88d84b7cc85841c575e54b635b125012c1d2e39d Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Sat, 11 Feb 2012 00:06:32 -0500
Subject: [PATCH] fix default nameserver when resolv.conf doesn't exist

---
 src/network/__dns.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/network/__dns.c b/src/network/__dns.c
index 45d9b43..0d4f47e 100644
--- a/src/network/__dns.c
+++ b/src/network/__dns.c
@@ -82,8 +82,9 @@ int __dns_doqueries(unsigned char *dest, const char *name, int *rr, int rrcnt)
 	}
 	if (f) __fclose_ca(f);
 	if (!nns) {
-		ns[0].sin.sin_family = AF_INET;
+		ns[0].sin.sin_family = family = AF_INET;
 		ns[0].sin.sin_port = htons(53);
+		ns[0].sin.sin_addr.s_addr = htonl(0x7f000001);
 		nns=1;
 		sl = sizeof sa.sin;
 	}
-- 
1.7.3.2
